cmd/tiller-proxy/proxy/proxy_test.go

```diff
@@ -0,0 +1,364 @@
+/*
+Copyright (c) 2018 Bitnami
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package proxy
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/helm/pkg/helm"
+	"k8s.io/helm/pkg/proto/hapi/chart"
+	"k8s.io/helm/pkg/proto/hapi/release"
+	"k8s.io/helm/pkg/repo"
+)
+
+// Fake server for repositories and charts
+type fakeHTTPClient struct {
+	repoURLs  []string
+	chartURLs []string
+	index     *repo.IndexFile
+}
+
+func (f *fakeHTTPClient) Do(h *http.Request) (*http.Response, error) {
+	for _, repoURL := range f.repoURLs {
+		if h.URL.String() == fmt.Sprintf("%sindex.yaml", repoURL) {
+			// Return fake chart index (not customizable per repo)
+			body, err := json.Marshal(*f.index)
+			if err != nil {
+				fmt.Printf("Error! %v", err)
+			}
+			return &http.Response{StatusCode: 200, Body: ioutil.NopCloser(bytes.NewReader(body))}, nil
+		}
+	}
+	for _, chartURL := range f.chartURLs {
+		if h.URL.String() == chartURL {
+			// Simulate download time
+			time.Sleep(100 * time.Millisecond)
+			// Fake chart response
+			return &http.Response{StatusCode: 200, Body: ioutil.NopCloser(bytes.NewReader([]byte{}))}, nil
+		}
+	}
+	// Unexpected path
+	return &http.Response{StatusCode: 404}, fmt.Errorf("Unexpected path")
+}
+
+func fakeLoadChart(in io.Reader) (*chart.Chart, error) {
+	// Fake I/O time
+	time.Sleep(100 * time.Millisecond)
+	return &chart.Chart{}, nil
+}
+
+func prepareTestProxy(hrs []helmRelease, existingTillerReleases []AppOverview) *Proxy {
```

Comment:
I would name this `newTestProxy` or `newFakeProxy` - `prepare` doesn't sound like something that returns a `*Proxy` to me.